src/loom-provider.ts

```diff
@@ -91,4 +92,48 @@ export class LoomProvider {
 
     if (ret) callback(null, ret)
   }
+
+  protected _callAsync(payload: any): Promise<any> {
+    const caller = new Address(this._client.chainId, LocalAddress.fromHexString(payload.from))
+    const address = new Address(this._client.chainId, LocalAddress.fromHexString(payload.to))
+    const ui8InData = this._hexStringToUint8Array(payload.data.substring(2))
+
+    const callTx = new CallTx()
+    callTx.setVmType(VMType.EVM)
+    callTx.setInput(ui8InData)
+
+    const msgTx = new MessageTx()
+    msgTx.setFrom(caller.MarshalPB())
+    msgTx.setTo(address.MarshalPB())
+    msgTx.setData(callTx.serializeBinary())
+
+    const tx = new Transaction()
+    tx.setId(2)
+    tx.setData(msgTx.serializeBinary())
+
+    return this._client.commitTxAsync<Transaction>(tx)
+  }
+
+  protected _callStaticAsync(payload: any): Promise<any> {
+    const address = new Address(this._client.chainId, LocalAddress.fromHexString(payload.to))
+    const ui8InData = this._hexStringToUint8Array(payload.data.substring(2))
+    return this._client.queryAsync(address, ui8InData, VMType.EVM)
+  }
+
+  protected _hexStringToUint8Array(hexInput:string): Uint8Array {
+    const bytes = new Uint8Array(Math.ceil(hexInput.length / 2))
+    for (let i = 0; i < bytes.length; i++) {
+      bytes[i] = parseInt(hexInput.substr(i * 2, 2), 16)
+    }
+    return bytes
+  }
+
+  protected _Uint8ArrayToHexString(data: Uint16Array): string {
```

Comment:
So is `data` a `Uint16Array`? Why is this called `Uint8ArrayToHexString` then?

Reply:
My bad I'll fix it